src/bug.rs

```diff
@@ -0,0 +1,309 @@
+//! `!bug` - A prioritized bug list for users.
+//!
+//! # Usage
+//!
+//! * `!bug [show, print, display]` - Print the bug list.
+//! * `!bug [show, print, display] <BUG_NUMBER>` - Print the details for a specific bug.
+//! * `!bug [report] <BUG_NAME> <BUG_SUMMARY> <BUG_DETAILS>` - Add a bug to the list.
+//! * `!bug (remove, rm, delete) <BUG_NUMBER>` - Remove a bug from the list.
+//! * `!bug +1 <BUG_NUMBER>` - Report that you've also encoutered this bug.
+
+use anyhow::{Context, Ok, Result};
+use mongodb::{bson::doc, Database};
+use pest::Parser;
+use serde::{Deserialize, Serialize};
+use serenity::model::prelude::{Message, UserId};
+use std::{collections::HashMap, fmt};
+use tracing::{debug, info};
+
+static COLLECTION_NAME: &str = "global_bugs";
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum BugStatus {
+    Open,
+    Closed,
+}
+
+impl fmt::Display for BugStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            BugStatus::Open => write!(f, "Open"),
+            BugStatus::Closed => write!(f, "Closed"),
+        }
+    }
+}
+
+impl Default for BugStatus {
+    fn default() -> Self {
+        BugStatus::Open
+    }
+}
+
+/// A global list of bugs.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct BugList {
+    /// The bugs in the global list. The key is the item key, and the value is the
+    /// item state.
+    items: HashMap<u32, BugItem>,
+}
+
+impl BugList {
+    fn new() -> Self {
+        BugList {
+            items: Default::default(),
+        }
+    }
+}
+
+/// A single bug item in a user's bug list.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct BugItem {
+    // Hopefully our code isn't so bad that we'll need more than this.
+    pub number: u32,
+    pub priority: u32,
+    pub status: BugStatus,
+    pub labels: Vec<String>,
+    pub name: String,
+    pub summary: String,
+    pub details: String,
+    pub reporter: UserId,
+    pub plus_ones: Vec<UserId>,
```

Comment:
I think let's also skip `plus_ones` for now. If we go in that direciton we probs wanna support a more generic emoji reaction system rather than hard coding the concept of "plus ones".